test/e2e-go/features/catchup/basicCatchup_test.go

```diff
@@ -121,3 +124,76 @@ func TestCatchupOverGossip(t *testing.T) {
 	err = fixture.LibGoalFixture.ClientWaitForRoundWithTimeout(lg, waitForRound)
 	a.NoError(err)
 }
+
+func TestStoppedCatchupOnUnsupported(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+	a := require.New(t)
+
+	os.Setenv("ALGORAND_TEST_UNUPGRADEDPROTOCOL_DELETE_CURRENT_VERSION", "0")
+
+	// Overview of this test:
+	// Start a two-node network (primary has 0%, secondary has 100%)
+	// Let it run for a few blocks.
+	// Spin up a third node and see if it catches up
+
+	var fixture fixtures.RestClientFixture
+	// Give the second node (which starts up last) all the stake so that its proposal always has better credentials,
+	// and so that its proposal isn't dropped. Otherwise the test burns 17s to recover. We don't care about stake
+	// distribution for catchup so this is fine.
+	fixture.Setup(t, filepath.Join("nettemplates", "TwoNodes100SecondTestUnupgradedProtocol.json"))
+	defer fixture.Shutdown()
+
+	// Get 2nd node so we wait until we know they're at target block
+	nc, err := fixture.GetNodeController("Node")
+	a.NoError(err)
+
+	// Let the network make some progress
+	a.NoError(err)
+	waitForRound := uint64(3) // UpgradeVoteRounds + DefaultUpgradeWaitRounds
+	err = fixture.ClientWaitForRoundWithTimeout(fixture.GetAlgodClientForController(nc), waitForRound)
+	a.NoError(err)
+
+	os.Setenv("ALGORAND_TEST_UNUPGRADEDPROTOCOL_DELETE_CURRENT_VERSION", "1")
+
+	// Now spin up third node
+	cloneDataDir := filepath.Join(fixture.PrimaryDataDir(), "../clone")
+	cloneLedger := false
+	err = fixture.NC.Clone(cloneDataDir, cloneLedger)
+	a.NoError(err)
+	cloneClient, err := fixture.StartNode(cloneDataDir)
+	a.NoError(err)
+
+	// Now, catch up
+	err = fixture.LibGoalFixture.ClientWaitForRoundWithTimeout(cloneClient, waitForRound)
+	a.NoError(err)
+
+	var status v1.NodeStatus
+
+	// Check the status
+	timeout := time.NewTimer(20 * time.Second)
+	timedOut := false
+	go func() {
+		<-timeout.C
+		timedOut = true
+	}()
+	for status, err = cloneClient.Status(); !timedOut && err == nil && // did not timeout
+		// and while the next round is not the next protocol upgrade:
+		(status.NextVersion == status.LastVersion || // next is not a new protocol, or
+			// next is a new protocol but,
+			(status.NextVersion != status.LastVersion &&
+				// the new protocol version is not the next round
+				status.LastRound+1 != status.NextVersionRound)); status, err = cloneClient.Status() {
+		time.Sleep(800 * time.Millisecond)
+	}
+	a.NoError(err)
+	status, err = cloneClient.Status()
+	// Stoppd at the first protocol
+	a.Equal("test-unupgraded-protocol", status.LastVersion)
+	// Next version is different (did not upgrade to it)
+	a.True(status.NextVersion != status.LastVersion)
```

Comment:
nit: it's better to use Equal or NotEqual here, as it will printout both expressions in case of a failure.